src/librustc_save_analysis/lib.rs

```diff
@@ -742,9 +742,19 @@ impl<'l, 'tcx> SaveContext<'l, 'tcx> {
             Res::Def(HirDefKind::Mod, def_id) => {
                 Some(Ref { kind: RefKind::Mod, span, ref_id: id_from_def_id(def_id) })
             }
-            Res::PrimTy(..)
+
+            Res::Def(HirDefKind::Macro(..), _)
+            | Res::Def(HirDefKind::ExternCrate, _)
+            | Res::Def(HirDefKind::ForeignMod, _)
+            | Res::Def(HirDefKind::LifetimeParam, _)
+            | Res::Def(HirDefKind::AnonConst, _)
+            | Res::Def(HirDefKind::Use, _)
+            | Res::Def(HirDefKind::Field, _)
+            | Res::Def(HirDefKind::GlobalAsm, _)
+            | Res::Def(HirDefKind::Impl, _)
+            | Res::Def(HirDefKind::Closure, _)
```

Comment:
Here and elsewhere in this PR... `#![feature(or_patterns)]` are one bootstrap-bump PR away (already inside a rollup) from being usable. Can we block this PR on that (should be ready in a few hours ideally) and make the cleanup here?

Reply:
Oh, awesome. I tried to use that here, but it gave me a warning.

Reply:
FMOI (for my own info): #70029